test/e2e/cancel-request/stream-cancel.test.ts

```diff
@@ -55,31 +52,37 @@ describe('streaming responses cancel inner stream after disconnect', () => {
     ['edge pages api', '/api/edge-api'],
     ['node pages api', '/api/node-api'],
   ])('%s', (_name, path) => {
+    beforeAll(async () => {
+      // Trigger compilation of the route so that compilation time does not
+      // factor into the actual test requests.
+      await next.fetch(path + '?compile')
+    })
+
     it('cancels stream making progress', async () => {
       // If the stream is making regular progress, then we'll eventually hit
       // the break because `res.destroyed` is true.
       await prime(path + '?write=25')
       const res = await next.fetch(path)
-      const i = +(await res.text())
-      expect(i).toBeWithin(1, 5)
+      const i = await res.text()
+      expect(i).toBeOneOf(['1', '2', '3', '4', '5'])
     }, 2500)
 
     it('cancels stalled stream', async () => {
       // If the stream is stalled, we'll never hit the `res.destroyed` break
       // point, so this ensures we handle it with an out-of-band cancellation.
       await prime(path + '?write=1')
       const res = await next.fetch(path)
-      const i = +(await res.text())
-      expect(i).toBe(1)
+      const i = await res.text()
+      expect(i).toBe('1')
     }, 2500)
 
     it('cancels stream that never sent data', async () => {
       // If the client has never sent any data (including headers), then we
       // haven't even established the response object yet.
       await prime(path + '?write=0', true)
       const res = await next.fetch(path)
-      const i = +(await res.text())
-      expect(i).toBe(0)
+      const i = await res.text()
+      expect(i).toBe('0')
```

Comment:
Removing the number coercions from the response text gives us better diffs when the expectation fails.